mypyc/ir/ops.py

```diff
@@ -1183,6 +1183,33 @@ def accept(self, visitor: 'OpVisitor[T]') -> T:
         return visitor.visit_call_c(self)
 
 
+class Truncate(RegisterOp):
```

Comment:
Add docstring (similar to other ops).

Mention that this only works between non-reference counted integer types or bool (not `int_rprimitive`).